pymongo/server_description.py

```diff
@@ -72,9 +76,11 @@ def __init__(
         self._error = error
         self._topology_version = hello.topology_version
         if error:
-            if hasattr(error, 'details') and isinstance(error.details, dict):
-                self._topology_version = error.details.get('topologyVersion')
+            details = getattr(error, 'details', None)
+            if isinstance(details, dict):
+                self._topology_version = cast(Any, error).details.get('topologyVersion')
```

Comment:
Let's remove the cast and replace error.details with "details" which is known to be a dict.

Reply:
Done